login-workflow/src/components/BrandedCardContainer/BrandedCardContainer.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { ReactNode } from 'react';
+import Card, { CardProps as CardPropsType } from '@mui/material/Card';
+import { SxProps, Theme, useTheme } from '@mui/material/styles';
+import { Spinner, SpinnerProps } from '../Spinner';
+import defaultBackgroundImage from '../../assets/images/background.svg';
+import { useInjectedUIContext } from '@brightlayer-ui/react-auth-shared';
+import Box, { BoxProps } from '@mui/material/Box';
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import {
+    getBrandedCardContainerUtilityClass,
+    BrandedCardContainerClassKey,
+    BrandedCardContainerClasses,
+} from './BrandedCardContainerClasses';
+import { cx } from '@emotion/css';
+
+export type BrandedCardContainerProps = BoxProps & {
+    loading?: boolean;
+    backgroundImage?: string;
+    cardStyles?: SxProps<Theme>;
+    CardProps?: CardPropsType;
+    LoaderComponent?: ReactNode;
+    classes?: BrandedCardContainerClasses;
+    slots?: { card?: React.ElementType; loader?: React.ElementType };
+    slotProps?: {
+        card?: CardPropsType;
+        loader?: SpinnerProps;
+    };
+};
+
+const useUtilityClasses = (ownerState: BrandedCardContainerProps): Record<BrandedCardContainerClassKey, string> => {
+    const { classes } = ownerState;
+
+    const slots = {
+        root: ['root'],
+        card: ['card'],
+    };
+
+    return composeClasses(slots, getBrandedCardContainerUtilityClass, classes);
+};
+
+/**
+ * Component that renders a card centered on a blue textured background.
+ *
+ * @param loading if true a loading spinner will be displayed blocking the content
+ *
+ * @param backgroundImage will be displayed in the background instead of `defaultBackgroundImage`
+ *
+ * @param cardStyles will override the styles for `Card` component
+ *
+ * @param CardProps will be the props passed to `Card` component
+ *
+ * @param LoaderComponent will be displayed instead of `Spinner`
+ *
+ * @param classes for default style overrides
+ *
+ * @category Component
+ */
+export const BrandedCardContainer: React.FC<BrandedCardContainerProps> = (props) => {
+    const {
+        children,
+        loading,
+        backgroundImage,
+        cardStyles,
+        CardProps,
+        slots = {},
+        slotProps = {},
+        LoaderComponent = <Spinner visible={loading} component={slots.loader} {...slotProps.loader} />,
```

Comment:
If you're providing a slot for `LoaderComponent` and I pass in my own `LoaderComponent` in the slots I expect it to render my `LoaderComponent`, which it doesn't.